source/api/commands/route2.md

```diff
@@ -0,0 +1,385 @@
+
+```javascript
+cy.route2('/users', (req) => {
+  // modify the headers or body
+  req.headers = {...req.headers, accept: 'application/json'}
+  req.body = {...req.body, foo: 'bar'}
+})
+```
+
+### Modify response
+
+After a request that gets passed through, we can modify the response from the server before it is returned through a function. The function exposes a `req` object which contains the request. The `req.send()` function exposes a `res` object which contains the response.
+
+```javascript
+cy.route2('/users', (req) => {
+  // passing a function to req.send causes the request to pass through
+  // and allows the response from the origin server to be modified
+  req.send((res) => {
+    res.status = 200
+    res.headers['x-new-headers'] = 'from-server'
+
+    // dynamically alter body
+    res.body = res.body.replace('window.top', 'window.self')
+  })
+})
+```
+
+## With Stubbing
+
+If you pass a `response` to `cy.route()`, Cypress will stub the response in the request.
+
+### `url` as a string
+
+When passing a `string` as the `url`, the URL must match *exactly* what you've written. You'll want to use the decoded string and not include any hash encoding (ie. use `@` instead of `%40`).
+
+```javascript
+cy.route2('https://localhost:7777/surveys/customer?email=john@doe.com', [
+  {
+    id: 1,
+    name: 'john'
+  }
+])
+```
+
+### `url` as a RegExp
+
+When passing a RegExp as the `url`, the url will be tested against the regular expression and will apply if it passes.
+
+```javascript
+cy.route2(/users\/\d+/, { id: 1, name: 'Phoebe' })
+```
+
+```javascript
+// Application Code
+$.get('https://localhost:7777/users/1337', (data) => {
+  console.log(data) // => {id: 1, name: "Phoebe"}
+})
+```
+
+### Response functions
+
+You can also use a function as a response which enables you to add logic and modify properties of the response. The function exposes a `req` object which contains the request as well as a way to `send` with a response.
+
+```javascript
+cy.route2('/login', (req) => {
+  req.send('Success')
+})
+```
+
+```javascript
+// reply with body and headers
+cy.route2('/login', (req) => {
+  req.send('Success', { 'x-new-headers': 'from-server' })
+})
+```
+
+```javascript
+// reply with an object containing more response details
+cy.route2('/login', (req) => {
+  req.send({
+    statusCode: 200,
+    body: JSON.stringify({
+      some: 'response'
+    }),
+    headers: {
+      'x-new-headers': 'from-server'
+    },
+    // If `destroySocket` is truthy, Cypress will destroy the connection to the
+    // browser and send no response. Useful for simulating a server that is not
+    // reachable. Must not be set in combination with other options.
+    destroySocket: false
+  })
+})
+```
+
+```javascript
+// continue the HTTP response to the browser, including any modifications made to `res`
+cy.route2('/login', (req) => {
+  req.send()
+})
+```
```

Comment:
this is functionally identical to the following code, but has worse performance because of sending req/res to the browser unnecessarily

```suggestion
```javascript
// continue the HTTP response to the browser, including any modifications made to `res` <--- what's `res` here?
cy.route2('/login')
```
```